cdap-ui/app/cdap/components/OpsDashboard/RunsGraph/DataParser.js

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright © 2018 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import moment from 'moment';
+import uniqWith from 'lodash/uniqWith';
+import isEqual from 'lodash/isEqual';
+
+export function parseDashboardData(data, startTime, duration) {
+  let {
+    buckets,
+    timeArray
+  } = setBuckets(startTime, duration);
+
+  let pipelineCount = 0,
+      customAppCount = 0;
+
+  data.forEach((runInfo) => {
+    if (['cdap-data-pipeline', 'cdap-data-streams'].indexOf(runInfo.artifact.name) !== -1) {
+      pipelineCount++;
+    } else {
+      customAppCount++;
+    }
+
+    let startTime = getBucket(runInfo.start);
+    let endTime = getBucket(runInfo.end);
+
+    // add start method
+    if (runInfo.startMethod === 'Manual') {
+      buckets[startTime].manual++;
+    } else {
+      buckets[startTime].schedule++;
+    }
+    buckets[startTime].runsList.push(runInfo);
+
+    // aggregate delay
+    let delay = runInfo.running - runInfo.start;
+    buckets[startTime].delay += delay;
+
+    // add status
+    if (endTime) {
+      if (runInfo.status === 'COMPLETED') {
+        buckets[endTime].successful++;
+      } else if (runInfo.status === 'FAILED') {
```

Comment:
More of a product question: We don't have to display runs that were 'KILLED'?

Reply:
Still unsure at the moment, this bucket aggregation still needs a bit of discussions